pkg/controllers/nodeclass/status/readiness.go

```diff
@@ -1,63 +0,0 @@
-/*
-Licensed under the Apache License, Version 2.0 (the "License");
-you may not use this file except in compliance with the License.
-You may obtain a copy of the License at
-
-    http://www.apache.org/licenses/LICENSE-2.0
-
-Unless required by applicable law or agreed to in writing, software
-distributed under the License is distributed on an "AS IS" BASIS,
-WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-See the License for the specific language governing permissions and
-limitations under the License.
-*/
-
-package status
-
-import (
-	"context"
-	"fmt"
-
-	"github.com/awslabs/operatorpkg/status"
-	"github.com/samber/lo"
-
-	"github.com/aws/karpenter-provider-aws/pkg/providers/launchtemplate"
-
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-
-	"github.com/aws/karpenter-provider-aws/pkg/apis/v1beta1"
-)
-
-type Readiness struct {
-	launchTemplateProvider launchtemplate.Provider
-}
-
-func (n Readiness) Reconcile(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) (reconcile.Result, error) {
```

Comment:
Can just not move these checks to each reconcile function of each resource? Instead of having a controller for each field we change